pkg/api/v1/types.go

```diff
@@ -1482,6 +1482,16 @@ type AWSNodeSpec struct {
 	AssignPublicIP *bool `json:"assignPublicIP"`
 	// IsSpotInstance indicates whether the created machine is an aws ec2 spot instance or on-demand ec2 instance.
 	IsSpotInstance *bool `json:"isSpotInstance"`
+	// SpotInstanceMaxPrice The maximum price you are willing to pay per instance hour. Your instance runs when
+	// your maximum price is greater than the Spot Price.
+	SpotInstanceMaxPrice *string `json:"spotInstanceMaxPrice"`
+	// SpotInstancePersistentRequest Ensures that your request will be submitted every time your Spot Instance is terminated.
+	// For information about Persistent requests and other customization options
```

Comment:
I think that you're missing the part of the comment here. Do you want to point to the AWS docs here?

Reply:
Nah, I will just omit it :D